react/components/AddressCompletionForm.tsx

```diff
@@ -62,7 +62,12 @@ const AddressCompletionForm: React.FC<Props> = ({
       updatedAddress.receiverName = `${firstName} ${lastName}`
```

Comment:
the issue above is because this line is mutating the address object in the context.. I think it could be solved if we cloned the object instead of just assigning it to the `updatedAddress` variable.

Reply:
This does not explain why the checkbox disappears when you type something in the input, right? Because the form is not being submitted at that time. I only managed to reproduce this once, and I don't even know how I did it...
Anyway, I changed this component so that the condition to show the checkbox uses the `selectedAddress` from the state. I think that should work as well.

Reply:
I could reproduce it consistently by pressing the submit button and then type something in the input. It kinda makes sense if you see that we are mutating the object that is being stored in the `AddressContextProvider` state.

So, when we type in the number field input and the `AddressForm` component updates the address with the new number field value ([here](https://github.com/vtex-apps/place-components/blob/master/react/AddressForm.tsx#L218-L221)), the `prevAddress` already has the receiver name because we mutated it, and because it is a `setState`, the `AddressCompletionForm` re-renders and the check `receiverName != null` is now `false`, which causes us to remove the checkbox

Reply:
Ah! Naruhodo.
I'll clone the `address` object here then to prevent mutating the context state object and probably causing more weird bugs in the future.